core/CMakeLists.txt

```diff
@@ -9,56 +9,63 @@ message(STATUS "Building 'detray::core' component")
-   "include/detray/definitions/cuda_defs.hpp"
+   # definitions include(s)
+   "include/detray/definitions/cuda_definitions.hpp"
    "include/detray/definitions/qualifiers.hpp"
-   "include/detray/definitions/invalid_values.hpp"
+   "include/detray/definitions/track_parameterization.hpp"
    "include/detray/definitions/units.hpp"
+   # field include(s)
    "include/detray/field/constant_magnetic_field.hpp"
-   "include/detray/geometry/index_geometry.hpp"
+   # geometry include(s)
    "include/detray/geometry/surface.hpp"
    "include/detray/geometry/volume_connector.hpp"
+   "include/detray/geometry/volume_graph.hpp"      
    "include/detray/geometry/volume.hpp"
-   "include/detray/grids/associator.hpp"
+   # grids include(s)
    "include/detray/grids/axis.hpp"
    "include/detray/grids/grid2.hpp"
    "include/detray/grids/populator.hpp"
    "include/detray/grids/serializer2.hpp"
+   # intersection include(s)
+   "include/detray/intersection/concentric_cylinder_intersector.hpp"
+   "include/detray/intersection/cylinder_intersector.hpp"
+   "include/detray/intersection/intersection_kernel.hpp"
+   "include/detray/intersection/intersection.hpp"
+   "include/detray/intersection/planar_intersector.hpp"
+   "include/detray/intersection/quadratic_equation.hpp"   
+   "include/detray/intersection/unbound.hpp"   
+   # masks include(s)
    "include/detray/masks/annulus2.hpp"
    "include/detray/masks/cylinder3.hpp"
    "include/detray/masks/masks.hpp"
    "include/detray/masks/rectangle2.hpp"
    "include/detray/masks/ring2.hpp"
    "include/detray/masks/single3.hpp"
    "include/detray/masks/trapezoid2.hpp"
-   "include/detray/masks/unmasked.hpp"
-   "include/detray/tools/base_stepper.hpp"
+   "include/detray/masks/unmasked.hpp"   
+   # propagator include(s)
+   "include/detray/propagator/base_stepper.hpp"
+   "include/detray/propagator/line_stepper.hpp"
+   "include/detray/propagator/navigator.hpp"
+   "include/detray/propagator/propagator.hpp"
+   "include/detray/propagator/rk_stepper.hpp"
+   "include/detray/propagator/track.hpp"   
+   # tools include(s)
+   "include/detray/tools/associator.hpp"
    "include/detray/tools/bin_association.hpp"
-   "include/detray/tools/concentric_cylinder_intersector.hpp"
-   "include/detray/tools/cylinder_intersector.hpp"
-   "include/detray/tools/volume_graph.hpp"
+   "include/detray/tools/generators.hpp"
    "include/detray/tools/grid_array_helper.hpp"
-   "include/detray/tools/intersection_kernel.hpp"
-   "include/detray/tools/line_stepper.hpp"
    "include/detray/tools/local_object_finder.hpp"
-   "include/detray/tools/navigator.hpp"
-   "include/detray/tools/planar_intersector.hpp"
-   "include/detray/tools/propagator.hpp"
-   "include/detray/tools/track.hpp"   
-   "include/detray/tools/track_parameterization.hpp"
-   "include/detray/tools/rk_stepper.hpp"
-   "include/detray/utils/c_step.hpp"
+   # utils include(s)
+   "include/detray/utils/detail/accessor.hpp"
```

Comment:
Honestly, I would have put ```accessor```, ```invalid_values```, ```indexing``` and especially ```containers``` into 'definitions', since they are setting global type definitions, even if there is a bit of implementation in there, too.

Reply:
OK. But let me leave `invalid_values` in utils at least..

Reply:
Sure :). I was initially under the impression that we wanted to merge 'tools' and 'utils', but I am OK with leaving it